### PR TITLE
Loginp3

### PR DESCRIPTION
**Pull request by smithatlanta**
Removed login calls from up, down, and generate but left guards in
place at api calls.  Less chance of calling login twice.

### CODE REVIEW

File: cmd/down.go [hunk 1]
```diff
@@ -27,11 +27,6 @@ func down(cmd *cobra.Command, args []string) {
 	//read the harbor compose file
 	var harborCompose = DeserializeHarborCompose(HarborComposeFile)
 
-	_, _, err := Login()
```
Comment (jritsema) [inline cmd/down.go:30]: I kinda like having `Login()` calls here in the cli layer and keeping the harborAPI code just a thin abstraction of the REST API that isn't aware of the login/logout cli semantics.

Can't the returned `user` and `token` be captured here and passed in to the harborAPI?

Comment (smithatlanta) [inline cmd/down.go:30]: If can be but it adds more baggage to the code to have to pass that around everywhere.  It's much more isolated the way it's coded now.

Comment (jkurz) [inline cmd/down.go:30]: I say just leave this as is. HarborApi relies on auth so i can see it go either way.

Comment (jritsema) [inline cmd/down.go:30]: I guess the scenario I was specifically thinking about was replacing the harborApi code with a shared harbor golang client library.  Wilson and I had talked about sharing a common library between Harbor Compose and the ShipIt CLI.  So if we end up replacing it, with this proposed design, we'll lose the CLI login behavior along with it and will have to add the code back to the cmd layer.

**Pull request merged**